core/src/main/java/org/jdbi/v3/core/qualifier/QualifiedType.java

```diff
@@ -73,9 +97,19 @@
         return (QualifiedType<T>) of(type.getType());
     }
 
-    private QualifiedType(Type type, Set<? extends Annotation> qualifiers) {
+    private QualifiedType(Type type, Object qualifiers) {
```

Comment:
could it be interesting to guard type safety against mistakes by having 3 overloaded constructors for the specific cases instead?

Reply:
see my other comment on the variable type